src/operators/openapi.operators.ts

```diff
@@ -0,0 +1,66 @@
+import 'reflect-metadata';
+import { Type } from '@nestjs/common';
+import { trimStart } from 'lodash';
+
+/* @nestjs/swagger does not expose its internal metadata operations, so we partially
+ * reproduce them here. Obviously, this code takes advantage of internal functionality
+ * and may be subject to regression if the underlying libray changes its implementation.
+ */
+
+/* The Reflect metadata key used to store the array of OpenAPI properties.
+ */
+const OPENAPI_PROPERTIES_ARRAY = 'swagger/apiModelPropertiesArray';
```

Comment:
`@nestjs/swagger` saves metadata for each property and also saves an array of property names. The simplest way to narrow one of its DTOs is to change the properties array.